src/main/java/org/takes/facets/auth/PsByFlag.java

```diff
@@ -98,19 +97,22 @@ public PsByFlag(final String flg, final Map<Pattern, Pass> map) {
     }
 
     @Override
-    public Iterator<Identity> enter(final Request req) throws IOException {
+    public Opt<Identity> enter(final Request req) throws IOException {
         final Iterator<String> flg = new RqHref.Base(req).href()
             .param(this.flag).iterator();
-        final Collection<Identity> users = new ArrayList<Identity>(1);
+        Opt<Identity> retValue = null;
         if (flg.hasNext()) {
             final String value = flg.next();
             for (final Map.Entry<Pattern, Pass> ent : this.passes.entrySet()) {
                 if (ent.getKey().matcher(value).matches()) {
-                    users.add(ent.getValue().enter(req).next());
+                    retValue = ent.getValue().enter(req);
```

Comment:
@antonini please add a check `if (user.has()) { break }` similiar to other Passes